src/tests/customer.test.ts

```diff
@@ -53,6 +53,23 @@ describe('customer', () => {
             )
         })
 
+        // fairly useless test as it will always pass, just used to checkout the generics functionality
+        it('retrieves data using a generic', async () => {
+            interface Campaign {
+                campaign: {
+                    resource_name: string,
+                    name: string
+                }
+            }
+
+            await customer.report<Campaign>({
```

Comment:
I know the test is not much use, but the correct type here should be `Campaign[]`